ArticleTemplates/assets/js/modules/cards.js

```diff
@@ -1,96 +1,45 @@
-    }
-
-    function articleCardsInserter(html) {
-        var relatedContentWrapper = document.getElementsByClassName('related-content__wrapper')[0];
-
-        if (relatedContentWrapper) {
-            if (!html) {
-                articleCardsFailed();
-            } else {
-                relatedContentWrapper.innerHTML = html;
-                snapToGrid();
-            }
+    function ready(config) {
+        if (!initialised) {
+            initialised = true;
+            setupGlobals();
         }
     }
 
-    function articleCardsFailed() {
-        var relatedContent = document.getElementsByClassName('related-content')[0];
-
+    function getRelatedContentPosition() {
+        var relatedContent = document.querySelector('.related-content');
         if (relatedContent) {
-            relatedContent.classList.add('related-content--has-failed');       
-        }
-    }
-
-    function snapToGrid() {
-        var relatedContentList = document.getElementsByClassName('related-content__list')[0];
-
-        if (relatedContentList) {
-            setUpFlipSnap(relatedContentList);
-            window.addEventListener('resize', util.debounce(onResize.bind(null, relatedContentList), 100));
-        }
-    }
-
-    function onResize(relatedContentList) {
-        if (flipSnap && flipSnap.destroy && relatedContentList) {
-            flipSnap.destroy();
-            relatedContentList.removeAttribute('style');
-            setUpFlipSnap(relatedContentList);
+            observer.observe(relatedContent, { attributes: true });
+            return util.getElementOffset(relatedContent);
         }
     }
 
-    function setUpFlipSnap(relatedContentList) {
-        var container = relatedContentList.parentNode,
-            containerStyle = container.currentStyle || window.getComputedStyle(container),
-            containerWidth = container.offsetWidth - parseInt(containerStyle.paddingRight.replace('px','')) - parseInt(containerStyle.paddingLeft.replace('px',''));
-
-        // add a class with the number of child items, so we can set the widths based on that 
-        relatedContentList.classList.add('related-content__list--items-' + relatedContentList.childElementCount);
-
-        if (relatedContentList.scrollWidth > containerWidth) {
-            flipSnap = flipSnap(relatedContentList);
-
-            // Android needs to be notified of touch start / touch end so article navigation can be disabled / enabled
-            if (GU.opts.platform === 'android') {
-                relatedContentList.addEventListener('touchstart', onTouchStart);
-                relatedContentList.addEventListener('touchend', onTouchEnd);
-            }
-        }            
-    }
-
-    function onTouchStart() {
-        window.GuardianJSInterface.registerRelatedCardsTouch(true);
+    function setRelatedContentHeight(height) {
+        document.querySelector('.related-content').style.height = height + 'px';
```

Comment:
Can we assign this `querySelector` to a variable `relatedContent` and  wrap this in a conditional... 

`if (relatedContent) { relatedContent.style.height = height + 'px'; }`